src/components/PostDetails.tsx

```diff
@@ -1,106 +1,137 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
+import { Post } from '../types/Post';
 import { NewCommentForm } from './NewCommentForm';
+import { getComments, addComment, deleteComment } from '../utils/fetchClient';
+import { Comment, CommentData } from '../types/Comment';
 
-export const PostDetails: React.FC = () => {
-  return (
-    <div className="content" data-cy="PostDetails">
-      <div className="content" data-cy="PostDetails">
-        <div className="block">
-          <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
-          </h2>
+type Props = {
+  currPost: Post | null;
+  modalWindow: boolean;
+  setModalWindow: (value: boolean) => void;
+};
 
-          <p data-cy="PostBody">
-            eveniet quo quis laborum totam consequatur non dolor ut et est
-            repudiandae est voluptatem vel debitis et magnam
-          </p>
-        </div>
+export const PostDetails: React.FC<Props> = ({
+  currPost,
+  modalWindow,
+  setModalWindow,
+}) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [btnLoader, setBtnLoader] = useState(false);
+  const [commentLoader, setCommentLoader] = useState(false);
 
-        <div className="block">
-          <Loader />
+  const onAdd = ({ name, email, body }: CommentData) => {
+    setBtnLoader(true);
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+    const postId = currPost?.id || 0;
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+    return addComment({ postId, name, email, body })
+      .then(newComment => setComments([...comments, newComment]))
+      .catch(err => {
+        setError(`Failed to add comment: ${err.message || 'Unknown error'}`);
+      })
```

Comment:
Consider providing user feedback when an error occurs during the comment addition process. This will help users understand that their action did not succeed and why.